packages/shared/src/introspectMachine.ts

```diff
@@ -191,31 +191,40 @@ export const introspectMachine = (machine: XState.StateNode) => {
       (transition.target as unknown as XState.StateNode[])?.forEach(
         (targetNode) => {
           nodeMaps[targetNode.id].sources.add(transition.eventType);
-        },
+        }
       );
       if (transition.cond && transition.cond.name) {
         if (transition.cond.name !== "cond") {
           guards.addEventToItem(
             transition.cond.name,
             transition.eventType,
-            node.path,
+            node.path
           );
         }
       }
 
       (transition.target as unknown as XState.StateNode[])?.forEach(
         (targetNode) => {
-          /** Pick up invokes */
-          targetNode.invoke?.forEach((service) => {
-            const serviceSrc = getServiceSrc(service);
-            if (typeof serviceSrc !== "string") return;
-            services.addEventToItem(
-              serviceSrc,
-              transition.eventType,
-              node.path,
-            );
+          /**
+           * We gather info on all the invocations that begin on
+           * all the nodes that are initial state nodes of the children
+           * of this node (see recursive-invoke)
+           */
+
+          const nodesToGather = [targetNode, ...targetNode.initialStateNodes];
```

Comment:
note: the logic in this file doesn't handle history nodes at all - so that should be fixed as well, but in a separate PR